src/SDIOAnalyzerResults.cpp

```diff
@@ -94,60 +95,126 @@ void SDIOAnalyzerResults::GenerateBubbleText( U64 frame_index, Channel& channel,
+    current_state = STATE_START;
 
-    U64 num_frames = GetNumFrames();
     for( U32 i = 0; i < num_frames; i++ )
     {
         Frame frame = GetFrame( i );
 
-        char time_str[ 128 ];
-        AnalyzerHelpers::GetTimeString( frame.mStartingSampleInclusive, trigger_sample, sample_rate, time_str, 128 );
+        switch( current_state )
+        {
+        case STATE_START:
+            AnalyzerHelpers::GetTimeString( frame.mStartingSampleInclusive, trigger_sample, sample_rate, value_str, 128 );
 
-        char number_str[ 128 ];
-        AnalyzerHelpers::GetNumberString( frame.mData1, display_base, 8, number_str, 128 );
+            file_stream << value_str;
 
-        file_stream << time_str << ",";
+            file_stream << "," << i;
 
-        if( frame.mType == SDIOAnalyzer::FRAME_DIR )
-        {
-            file_stream << "DIR:";
+            if( frame.mType != SDIOAnalyzer::FRAME_DIR )
+            {
+                current_state = STATE_ERR;
+                break;
+            }
+
+            current_state = STATE_DIR;
+            break;
+
+        case STATE_DIR:
+            if( frame.mType != SDIOAnalyzer::FRAME_DIR )
+            {
+                current_state = STATE_ERR;
+                break;
+            }
             if( frame.mData1 )
             {
-                file_stream << "from Host";
+                file_stream << ",C";
             }
             else
             {
-                file_stream << "from Slave";
+                file_stream << ",R";
             }
-        }
-        else if( frame.mType == SDIOAnalyzer::FRAME_CMD )
-        {
-            file_stream << "CMD:" << number_str;
-        }
-        else if( frame.mType == SDIOAnalyzer::FRAME_ARG )
-        {
-            file_stream << "ARG:" << number_str;
-        }
-        else if( frame.mType == SDIOAnalyzer::FRAME_LONG_ARG )
-        {
-            file_stream << "LONG_ARG:" << number_str;
-        }
-        else if( frame.mType == SDIOAnalyzer::FRAME_CRC )
-        {
-            file_stream << "CRC:" << number_str;
-        }
+            current_state = STATE_CMD;
+            break;
 
-        file_stream << std::endl;
+        case STATE_CMD:
+            if( frame.mType != SDIOAnalyzer::FRAME_CMD )
+            {
+                current_state = STATE_ERR;
+                break;
+            }
+            AnalyzerHelpers::GetNumberString( frame.mData1, Decimal, 6, value_str, 128 );
+            file_stream << "," << value_str;
+            current_state = STATE_ARG;
+            break;
 
-        if( UpdateExportProgressAndCheckForCancel( i, num_frames ) == true )
-        {
-            file_stream.close();
-            return;
+        case STATE_ARG:
+            if( frame.mType == SDIOAnalyzer::FRAME_ARG )
+            {
+                AnalyzerHelpers::GetNumberString( frame.mData1, display_base, 32, value_str, 128 );
+                file_stream << "," << value_str;
+                file_stream << ",";
```

Comment:
is this additional comma intentional?

Reply:
It is, the other one is ARG2 for long response frames (R2).

Reply:
got it.